plugins/devtools-backend/src/service/router.ts

```diff
@@ -33,24 +33,38 @@ import { errorHandler } from '@backstage/backend-common';
 import express from 'express';
 import { getBearerTokenFromAuthorizationHeader } from '@backstage/plugin-auth-node';
 import { createPermissionIntegrationRouter } from '@backstage/plugin-permission-node';
+import { SignalService } from '@backstage/plugin-signals-node';
 
 /** @public */
 export interface RouterOptions {
   devToolsBackendApi?: DevToolsBackendApi;
   logger: Logger;
   config: Config;
   permissions: PermissionEvaluator;
+  signalService?: SignalService;
 }
 
 /** @public */
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, config, permissions } = options;
+  const { logger, config, permissions, signalService } = options;
 
   const devToolsBackendApi =
     options.devToolsBackendApi || new DevToolsBackendApi(logger, config);
 
+  if (signalService) {
+    // Publish info periodically using the signal service
+    setInterval(async () => {
+      if (signalService.hasSubscribers('devtools:resources')) {
```

Comment:
Not sure if this is really a pattern that we can start to rely on tbh. Maybe it's something we can perhaps build into the signals client itself at some point? I'm pretty confident that we should constrain the design here in such as way that signals must always be extremely cheap - i.e. no backend plugin should want to avoid sending a signal to save on resources, but instead have it be very cheap to fire it off and forget about.

Reply:
Removed this